contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,210 @@
+However, if the app allows users to change data through the view (e.g., entering your name in the textbox), the developer has to write code to propagate that change to the Model.
+
+In contrast, Angular provides data binding in both ways. That means even changes to the View can be propagated to the Model automatically, without writing extra code for that.
+
+This [article](https://medium.com/@preethi.s/angular-custom-two-way-data-binding-3e618309d6c7) on Medium provides a good introduction to two-way data binding in Angular.
+
+### Disadvantages of Angular.
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning curve** - Angular requires you to learn many concepts, such as directives, modules, components, services and many more. 
+The large number of new concepts can be confusing to newcomers. Additionally, Angular also requires you to learn a new language, TypeScript.
+
+2. **Opinionated Framework** - Angular is opinionated about how you structure your code. This means that Angular provides you defaults for building applications. For instance, there are in-built services to perform data fetching, state management and much more. It is also opinionated about the development language as it requires you to use TypeScript.<br/>
+In contrast, React allows you to integrate any third party services in your application.
+
+You can refer to this [article](https://jsreport.io/why-is-react-more-popular-than-angular/) that points out the benefits other tools like React have over Angular.
+    
+### Angular when compared with other popular frontend frameworks
+
+[React]("https://reactjs.org/") and [Vue]("https://vuejs.org/") are two popular alternatives to Angular.
+
+React is a popular Javascript library, open sourced by Facebook. The ease of learning React is a key advantage.
+It also provides more flexibility as it allows integration of third party libraries.
+Similarly, Vue offers some advantages, like flexibility, simple structure and ease of integration.
+
+Every framework has its own pros and cons. The framework you choose depends on the requirements of the application. Below are a few resources that give a brief comparison between the popular frontend frameworks.
+
+1. [React vs Angular vs Vue](https://medium.com/@TechMagic/reactjs-vs-angular5-vs-vue-js-what-to-choose-in-2018-b91e028fa91d) - A brief comparison between the three most popular frontend tools.
+2. [Why use Vue](https://medium.com/@brainmobi/advantages-of-using-vue-js-for-your-web-applications-7e460cadfffc) - Advantages of using Vue.
+3. [Angular vs React](https://programmingwithmosh.com/react/react-vs-angular/) - A comprehensive comparison between React and Angular.
+
+## How to get started with Angular?
+
+Angular is not the easiest framework to work with. Given below are a few steps that will help you ease into the Angular environment.
```

Comment:
`Angular is not the easiest framework to work with.` You've mentioned this once already. No need to repeat the negative aspects :-p